chains/evm/listener/event-handler.go

```diff
@@ -112,35 +133,49 @@ func Erc20EventHandler(sourceID, destId uint8, nonce uint64, handlerContractAddr
+	type GenericHandlerDepositRecord struct {
+		DestinationChainID uint8
+		Depositer          common.Address
+		ResourceID         [32]byte
+		MetaData           []byte
 	}
-	res, err := a.Unpack("getDepositRecord", out)
+
+	rec, err := getDepositRecord(definition, nonce, destId, &handlerContractAddress, client)
 	if err != nil {
 		return nil, err
 	}
-	if len(res) == 0 {
-		return nil, errors.New("no handler associated with such resourceID")
-	}
 
-	out0 := *abi.ConvertType(res[0], new(ERC20HandlerDepositRecord)).(*ERC20HandlerDepositRecord)
+	out0 := *abi.ConvertType(rec, new(GenericHandlerDepositRecord)).(*GenericHandlerDepositRecord)
```

Comment:
But you should not be removing ERC20) Generic should be added as the new one

Reply:
I think git diff is weird on this, because the file contains both handlers

Reply:
so weird